sqlitecluster/SQLiteNode.cpp

```diff
@@ -1823,8 +1815,16 @@ void SQLiteNode::_changeState(SQLiteNodeState newState) {
     _localCommitNotifier.notifyThrough(_db.getCommitCount());
 
     if (newState != _state) {
-        // First, we notify all plugins about the state change
-        _server.notifyStateChangeToPlugins(*pluginDB, newState);
+        {
+            // We get a new handle here so that the sync node can operate on its handle and the plugin gets
+            // its own handle to operate on. This avoids conflicts where the sync thread and the plugin are trying to
+            // both run queries at the same time. This also avoids the need to create any share locking between the two.
+            SQLiteScopedHandle dbScope(*_dbPool, _dbPool->getIndex());
+            SQLite& pluginDB = dbScope.db();
+
+            // First, we notify all plugins about the state change
+            _server.notifyStateChangeToPlugins(pluginDB, newState);
+        }
```

Comment:
the handle still exists in memory since its in the pool. the more important thing is we don't have a handle sitting around unused, we just use an existing one that is not in use.